sat/cli/bootprep/input/session_template.py

```diff
@@ -166,12 +137,7 @@ def validate_image_exists(self, **_):
         Raises:
             InputItemValidateError: if the image cannot be verified to exist
         """
-        # First check if the image is being created anew by the same input file
```

Comment:
Addressed with c9ae744a7f4666cd5330a09162ddd416d655b1dd

Reply:
I discussed this with Eli this morning. Here's a summary of our discussion.

This issue only happens during dry run. Here is an example of running `sat bootprep run --dry-run` against Eli's third example:

```
ncn-m001:~/haasken/bootprep # sat bootprep run --dry-run eli-example3.yaml
INFO: Validating given input file eli-example3.yaml
INFO: Input file successfully validated against schema
INFO: Given input did not define any CFS configurations
WARNING: Found 2 public keys in IMS whose contents match the contents of public key file /root/.ssh/id_rsa.pub. Using the first one, but you may wish to clean up the duplicates.
INFO: Using IMS public key with id 12873d8a-051f-4ef0-9f1e-324a6eb5112c
INFO: Found IMS base for image at index 0: recipe with name=cray-shasta-csm-sles15sp2-barebones.x86_64-shasta-1.5
INFO: Of the 1 that would be created, 1 have no dependencies and would be created first.
INFO: Would create 1 images.
INFO: Dry run, not creating images.
ERROR: No image with name elis_image exists for use by BOS session template at index 0.
ERROR: The BOS session template at index 0 is not valid. See errors above.
ERROR: One or more items is not valid in collection of BOS session templates
ERROR: The collection of BOS session templates is not valid. See errors above.
```

Note that this error does not occur if we are not in dry-run mode. The session template validation occurs after `sat.cli.bootprep.image.create_images` is called. When not in dry-run mode, by the time we get to session template validation, the image will have actually been created, and so the session template will validate. E.g.:

```
ncn-m001:~/haasken/bootprep # sat bootprep run  eli-example3.yaml
INFO: Validating given input file eli-example3.yaml
INFO: Input file successfully validated against schema
INFO: Given input did not define any CFS configurations
WARNING: Found 2 public keys in IMS whose contents match the contents of public key file /root/.ssh/id_rsa.pub. Using the first one, but you may wish to clean up the duplicates.
INFO: Using IMS public key with id 12873d8a-051f-4ef0-9f1e-324a6eb5112c
INFO: Found IMS base for image at index 0: recipe with name=cray-shasta-csm-sles15sp2-barebones.x86_64-shasta-1.5
INFO: Of the 1 that will be created, 1 have no dependencies and will be created first.
INFO: Creating 1 images.
INFO: Creating images
INFO: Launching IMS job to create image
INFO: Created IMS image creation job with ID 743e6f56-9b01-4804-890d-2d01c1c070c0
INFO: Creation of image elis_image succeeded: ID 04eda969-f279-48b6-b934-ddfd3267b8bf
INFO: Image elis_image does not need configuration.
INFO: Creation of image elis_image succeeded: ID 04eda969-f279-48b6-b934-ddfd3267b8bf
INFO: Image creation completed successfully
INFO: Creating 1 BOS session template
```

After talking with Eli, I still think it's a valid concern that we should continue to allow session template validation to work in dry-run mode, and I think Eli's fix is correct. I'm going to test a little bit further with that fix. My main concern would be whether it's impacted at all by the possibility that image names can now be rendered using the base image or recipe. I think it will be fine, but I'm going to test that.